be/src/loan/service/loan.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { UserService } from "src/user/service/user.service";
+import { Repository } from "typeorm";
+import { UpdateLoanDTO } from "../dto/loan.update.dto";
+import { Loan } from "../entities/loan.entity";
+
+@Injectable()
+export class LoanService {
+  constructor( 
+    @InjectRepository(Loan) private readonly loanRepository: Repository<Loan>,
+    private readonly userService: UserService,
+  ) {}
+
+    async createOrUpdateLoan(id: number, updateLoan: UpdateLoanDTO, userId: number): Promise<Loan> {
+      const user = await this.userService.findById(userId);
+      const loan = await this.findById(id) || await this.loanRepository.create();
+      return await this.loanRepository.save({ ...loan, ...updateLoan, user });
+    }
```

Comment:
1. This function should be separated into two.
2. This lacks error handling. (if user doesn't exist for example we get an ugly err) [err handling example](https://github.com/TPT-Loane/TPT-Loane/blob/main/be/src/category/category.service.ts#L56)
3. I think updating the user should be an entirely other mutation (`updateLoanUser`), or maybe it even shouldn't be update-able (only remove loan and then create new one for new user)